src/go/rpk/pkg/cli/cloud/login.go

```diff
@@ -237,3 +252,67 @@ func loginProfileFlow(ctx context.Context, fs afero.Fs, y *config.RpkYaml, auth
 	}
 	return msg, y.Write(fs)
 }
+
+func clusterList(ctx context.Context, cl *cloudapi.Client) (vcs []cloudapi.VirtualCluster, cs []cloudapi.Cluster, err error) {
+	g, egCtx := errgroup.WithContext(ctx)
+	g.Go(func() (rerr error) {
+		vcs, rerr = cl.VirtualClusters(egCtx)
+		if rerr != nil {
+			return fmt.Errorf("unable to get the list of virtual clusters: %v", rerr)
+		}
+		return nil
+	})
+	g.Go(func() (rerr error) {
+		cs, rerr = cl.Clusters(egCtx)
+		if rerr != nil {
+			return fmt.Errorf("unable to get the list of clusters: %v", rerr)
+		}
+		return nil
+	})
+	if err := g.Wait(); err != nil {
+		return nil, nil, err
+	}
+	return
+}
+
+type nameAndCluster struct {
+	name string
+	c    *cloudapi.Cluster
+	vc   *cloudapi.VirtualCluster
+}
+
+// clusterNames combines the names of Virtual Clusters and Clusters, sorted
+// alphabetically, and returns a list of nameAndCluster structs representing the
+// combined clusters (VClusters first, then Clusters).
+func clusterNames(vcs []cloudapi.VirtualCluster, cs []cloudapi.Cluster, nsIDToName map[string]string) []nameAndCluster {
```

Comment:
IMO it wouldn't hurt to have a wordy name that captures what is happening in this function to improve readability. Ex combineAndSortClusterNames. That said this may be a violation of house style so I'll defer to @twmb on it 😅